GameOfRunes/GameOfRunes/GameEngine/RemoveDelegate.swift

```diff
@@ -41,17 +41,27 @@ class RemoveDelegate {
         }
     }
 
-    func removeEnemy(_ entity: EnemyEntity, shouldDecreasePlayerHealth: Bool = false) {
+    func removeEnemy(_ entity: EnemyEntity, shouldDecreasePlayerHealth: Bool = false,
+                     shouldIncreaseScore: Bool = false) {
         removeEnemyFromGame(entity)
         
         if shouldDecreasePlayerHealth {
             gameEngine?.decreasePlayerHealth()
+        } else {
+            gameEngine?.incrementCombo()
+        }
+        
+        if shouldIncreaseScore {
+            guard let scoreComponent = entity.component(ofType: ScoreComponent.self) else {
+                return
+            }
+            gameEngine?.addScore(by: scoreComponent.scorePoints)
         }
 
         guard let gestureEntity = entity.gestureEntity else {
             return
         }
-        
+        _ = entity.removeGesture()
```

Comment:
Is this necessary? Wouldnt `gameEngine?.remove` handle it?

Reply:
Yeah this is removing the gestureEntity from the EnemyEntity. I think `gameEngine?.remove` removes only components from entities.